domain/port/service/service.go

```diff
@@ -78,8 +81,17 @@ func (s *Service) GetUnitOpenedPorts(ctx context.Context, unitUUID coreunit.UUID
 	return s.st.GetUnitOpenedPorts(ctx, unitUUID)
 }
 
-// GetMachineOpenedPorts returns the opened ports for all the units on the machine.
-// Opened ports are grouped first by unit name and then by endpoint.
+// GetAllOpenedPorts returns the opened ports in the model, grouped by unit name.
+//
+// NOTE: We do not group by endpoint here. It is not needed. Instead, we just
+// group by unit name
```

Comment:
May be a bit out of scope of this PR, but its a little bit confusing that we have a type `GroupedPortRanges` that is grouped by sometimes endpoints and sometimes units. It would be nicer if there were two types, `PortRangesByEndpoint` and `PortRangesByUnit`. Especially since they are both strings. It feels like a nice place for some bugs to come in, or at least time wasted in the future.

Reply:
Agree. Unfortunately that would be quite a substantial change. We use `GroupedPortRanges` in a lot of places, and group be a few different keys.

So for now I've settled with what we've got.

In future, we plan to move our core types into the domain. This would probably be the best time for this refactor

Reply:
Add this to the epic or this risk register